sdk/javascript/test/BaseValue_spec.js

```diff
@@ -0,0 +1,123 @@
+		expect(() => new BaseValue(1, 0x100)).to.throw('outside of valid 8-bit range');
+
+		expect(() => new BaseValue(2, -1)).to.throw('outside of valid 16-bit range');
+		expect(() => new BaseValue(2, 0x10000)).to.throw('outside of valid 16-bit range');
+
+		expect(() => new BaseValue(4, -1)).to.throw('outside of valid 32-bit range');
+		expect(() => new BaseValue(4, 0x100000000)).to.throw('outside of valid 32-bit range');
+
+		expect(() => new BaseValue(8, -BigInt(1))).to.throw('outside of valid 64-bit range');
+		expect(() => new BaseValue(8, BigInt('0x10000000000000000'))).to.throw('outside of valid 64-bit range');
+	});
+
+	it('can create unsigned base value', () => {
+		// Assert:
+		// - 8-bit
+		canCreateUnsignedBaseValue(0, 1, 0);
+		canCreateUnsignedBaseValue(0xFF, 1, 0xFF);
+
+		// - 16-bit
+		canCreateUnsignedBaseValue(0, 2, 0);
+		canCreateUnsignedBaseValue(0xFFFF, 2, 0xFFFF);
+
+		// - 32-bit
+		canCreateUnsignedBaseValue(0, 4, 0);
+		canCreateUnsignedBaseValue(0xFFFFFFFF, 4, 0xFFFFFFFF);
+
+		// - 64-bit
+		canCreateUnsignedBaseValue(BigInt(0), 8, BigInt(0));
+		canCreateUnsignedBaseValue(BigInt('0xFFFFFFFFFFFFFFFF'), 8, BigInt('0xFFFFFFFFFFFFFFFF'));
+	});
+
+	it('cannot create signed with values outised range', () => {
+		// Assert:
+		expect(() => new BaseValue(1, -0x81, true)).to.throw('outside of valid 8-bit range');
+		expect(() => new BaseValue(1, 0x80, true)).to.throw('outside of valid 8-bit range');
+
+		expect(() => new BaseValue(2, -0x8001, true)).to.throw('outside of valid 16-bit range');
+		expect(() => new BaseValue(2, 0x8000, true)).to.throw('outside of valid 16-bit range');
+
+		expect(() => new BaseValue(4, -0x80000001, true)).to.throw('outside of valid 32-bit range');
+		expect(() => new BaseValue(4, 0x80000000, true)).to.throw('outside of valid 32-bit range');
+
+		expect(() => new BaseValue(8, -BigInt('0x8000000000000001'), true)).to.throw('outside of valid 64-bit range');
+		expect(() => new BaseValue(8, BigInt('0x8000000000000000'), true)).to.throw('outside of valid 64-bit range');
+	});
+
+	it('can create signed base value', () => {
+		// Assert:
+		// - 8-bit
+		canCreateSignedBaseValue(-0x80, 1, -0x80);
+		canCreateSignedBaseValue(0x7F, 1, 0x7F);
+
+		// - 16-bit
+		canCreateSignedBaseValue(-0x8000, 2, -0x8000);
+		canCreateSignedBaseValue(0x7FFF, 2, 0x7FFF);
+
+		// - 32-bit
+		canCreateSignedBaseValue(-0x80000000, 4, -0x80000000);
+		canCreateSignedBaseValue(0x7FFFFFFF, 4, 0x7FFFFFFF);
+
+		// - 64-bit
+		canCreateSignedBaseValue(-BigInt('0x8000000000000000'), 8, -BigInt('0x8000000000000000'));
+		canCreateSignedBaseValue(BigInt('0x7FFFFFFFFFFFFFFF'), 8, BigInt('0x7FFFFFFFFFFFFFFF'));
+	});
+
+	it('can create via converter', () => {
+		// Act:
+		const val = MockValue.deserialize(new Uint8Array([0x23, 0x01, 0xEF, 0xCD, 0xAB, 0x89, 0x67, 0x45]));
+
+		// Assert:
+		expect(val.size()).to.equal(8);
+		expect(val.value).to.equal(DEFAULT_VALUE);
+	});
```

Comment:
unsure what you're trying to test here ?

Reply:
so, I do realize the test looks fake, what I want all basevalues to do is to do deserialize like mock value above
(so via `converter.bytesToInt` )

any suggestions?

Reply:
the real deserialize function will have to be _somewhere_. i'd just test that instead.